src/js/rps.js

```diff
@@ -0,0 +1,105 @@
+const rock = document.getElementById("rock");
+const paper = document.getElementById("paper");
+const scissor = document.getElementById("scissor");
+const rockIcon = "✊";
+const paperIcon = "✋";
+const scissorIcon = "✌️";
+let roundWinner = "";
+let playerScore = 0,
+  computerScore = 0;
+rock.addEventListener("click", () => game("rock", rockIcon));
+paper.addEventListener("click", () => game("paper", paperIcon));
+scissor.addEventListener("click", () => game("scissor", scissorIcon));
+
+const playerPoint = document.getElementById("playerScore");
+const computerPoint = document.getElementById("computerScore");
+const result = document.querySelector(".result");
+const info = document.querySelector(".info");
+const playerSign = document.getElementById("player-sign");
+const computerSign = document.getElementById("computer-sign");
+
+const gameOverModal = document.getElementById("game-over-modal");
+const gameOverResult = document.getElementById("game-over-result");
+const close = document.querySelector(".close");
+const playAgain = document.getElementById("play-again-btn");
+
+playAgain.addEventListener("click", () => {
+  location.reload();
+});
+close.addEventListener("click", () => {
+  gameOverModal.style.display = "none";
+});
+
+function game(playerSelection, icon) {
+  if (playerScore == 5 || computerScore == 5) {
+    rock.removeEventListener("click", () => game("rock"));
+    paper.removeEventListener("click", () => game("paper"));
+    (playerScore = 0), (computerScore = 0);
+    return;
+  }
+  playerSign.textContent = icon;
+  let computerSelection = getComputerChoice();
+  let winner = playRound(playerSelection, computerSelection);
+  playerPoint.textContent = "Player: " + playerScore;
+  computerPoint.textContent = "Computer: " + computerScore;
+  if (playerScore == 5) {
+    gameOverResult.textContent = "win!";
+    gameOverModal.style.display = "block";
+    return;
+  }
+  if (computerScore == 5) {
+    gameOverResult.textContent = "lose!";
+    gameOverModal.style.display = "block";
+    return;
+  }
+}
+
+function getComputerChoice() {
```

Comment:
I probably would split this functionality into two pieces, determining the computer choice and then applying a class to choose the icon.

That way the function for determining choice can be simplified:

```javascript
function generateComputerChoice() {
  const choices = ["rock", "paper", "scissors"];
  const randomIndex = Math.floor(Math.random() * choices.length);
  return choices[randomIndex];
}
```

Reply:
This approach would also allow the ingestion of the `outcomes` map described below, using array manipulation.